- name: Check if we have shade node in groups
  hosts: localhost
  tags: shade
  gather_facts: yes
  any_errors_fatal: true
  tasks:
      - name: Adding host to 'shade' group
        vars:
          shade_group: "{{ groups.hypervisor|default(groups.undercloud) }}"
        add_host:
          name: "{{ install.get('shade', {}).host|default(shade_group|first) }}"
          groups: "shade"

      - include_role:
            name: inventory-update
        vars:
            inventory_file_name: 'hosts-install'

#TODO: Remove workaround below when https://bugzilla.redhat.com/show_bug.cgi?id=1689875 is fixed
#When a package update triggeres a delete of /usr/bin/python symlink, shade ansible module fails to find the interpreter
- name: Update alternative for /usr/bin/python
  hosts: shade
  become: yes
  shell: |
      [ -e /usr/bin/python ] || alternatives --set python /usr/bin/python3
      [ -e /usr/bin/python3 ] ||  dnf install -y python36 || dnf install -y python37 || dnf install -y python || dnf install -u /usr/bin/python3
  ignore_errors: yes
  when: ansible_distribution_major_version == '8'

- name: Prepare shade node
  hosts: shade
  tags: shade
  gather_facts: yes
  become: yes
  any_errors_fatal: true
  roles:
      - role: shade
